pkg/providers/launchtemplate/launchtemplate.go

```diff
@@ -355,6 +360,12 @@ func (p *Provider) getInstanceProfile(ctx context.Context, nodeClass *v1beta1.EC
 	if nodeClass.Spec.InstanceProfile != nil {
 		return aws.StringValue(nodeClass.Spec.InstanceProfile), nil
 	}
+	if nodeClass.Spec.Role != "" {
+		if nodeClass.Status.InstanceProfile == "" {
+			return "", cloudprovider.NewNodeClassNotReadyError(fmt.Errorf("instance profile hasn't resolved for role"))
```

Comment:
Generally I think it should be easy to add this check in for subnets, security groups, and AMIs. Thoughts on adding this in here or just opting for fast follow?

Reply:
Agreed. I think it's out of the scope of this PR so it makes sense as a fast-follow.